sematic/types/types/list.py

```diff
@@ -110,17 +111,20 @@ def list_to_json_encodable_summary(
         get_json_encodable_summary(item, element_type) for item in value
     ]
 
-    element_char_len = None
+    max_item = 0
     if len(complete_summary) > 0:
-        element_char_len = len(json.dumps(complete_summary[0]))
-
-    estimated_total_char_len = (
-        (0 if element_char_len is None else len(value) * element_char_len)
-        + 2
-        + 2 * len(value)
-    )
-
-    max_item = math.floor((1000 / estimated_total_char_len))
+        # this logic doesn't account for the "length" and "summary"
+        # keys or outer wrapping brackets, but those represent roughly
+        # 1 thousandth of the payload in the max case, and thus can
+        # be ignored
+        max_element_byte_len = max(
+            len(json.dumps(element).encode("utf8")) for element in complete_summary
+        )
+        max_element_byte_len += 2  # for the comma and space between elements
+        max_elements = int(MAX_SUMMARY_BYTES / max_element_byte_len)
+        max_item = max(
```

Comment:
This would create the impression the list has fewer elements than it does have.
Is it possible to render a `, ...` at the end of the list in place of the elided elements?

Reply:
the front-end already does this. What we return as a summary contains both a sample of elements and an int with the total number of elements. Here's an example of how it renders on the `main` branch (I've redacted the string because it comes from a user).
<img width="653" alt="Screen Shot 2022-10-19 at 3 19 19 PM" src="https://user-images.githubusercontent.com/7181589/196815560-d77f2fd5-b771-4243-a383-689cfb2f9f43.png">